programs/mango-v4/src/instructions/account_settle_fees_accrued_with_mngo.rs

```diff
@@ -0,0 +1,98 @@
+use anchor_lang::prelude::*;
+use checked_math as cm;
+use fixed::types::I80F48;
+
+use crate::accounts_zerocopy::*;
+use crate::error::MangoError;
+use crate::state::*;
+
+use crate::accounts_ix::*;
+
+pub fn account_settle_fees_accrued_with_mngo(
+    ctx: Context<AccountSettleFeesAccruedWithMngo>,
+    max_settle: u64,
+) -> Result<()> {
+    // Cannot settle with yourself
+    require_keys_neq!(
+        ctx.accounts.account.key(),
+        ctx.accounts.dao_account.key(),
+        MangoError::CannotSettleWithSelf
+    );
+
+    let group = ctx.accounts.group.load()?;
+
+    let mut mngo_bank = ctx.accounts.mngo_bank.load_mut()?;
+    let mut settle_bank = ctx.accounts.settle_bank.load_mut()?;
+
+    let mut account = ctx.accounts.account.load_full_mut()?;
+    // account constraint #1
+    require!(
+        account.fixed.is_owner_or_delegate(ctx.accounts.owner.key()),
+        MangoError::SomeError
+    );
+    let mut dao_account = ctx.accounts.dao_account.load_full_mut()?;
+
+    // check positions exist, for nicer error messages
+    {
+        account.token_position(mngo_bank.token_index)?;
+        account.token_position(settle_bank.token_index)?;
+        dao_account.token_position(mngo_bank.token_index)?;
+        dao_account.token_position(settle_bank.token_index)?;
+    }
+
+    let discount_rate = I80F48::from_num::<f32>(group.fees_mngo_discount_rate);
+    let now_ts = Clock::get()?.unix_timestamp.try_into().unwrap();
+
+    let mut max_settle = I80F48::from_num::<u64>(max_settle.min(account.fixed.fees_accrued));
+
+    // if mngo token position has borrows, skip settling
+    let account_mngo_token_position = account.token_position_mut(mngo_bank.token_index)?.0;
+    let account_mngo_native = account_mngo_token_position.native(&mngo_bank);
+    if account_mngo_native.is_negative() {
+        msg!(
+            "mngo token position ({} native mngo) has borrows, nothing will be settled",
+            account_mngo_native
+        );
+        return Ok(());
+    }
+    let mngo_oracle_price = mngo_bank.oracle_price(
+        &AccountInfoRef::borrow(&ctx.accounts.mngo_oracle.as_ref())?,
+        None,
+    )?;
+    // mngo is exchanged at a discount
+    let mut max_settle_mngo = cm!(max_settle * discount_rate / mngo_oracle_price);
+    // settlement is restricted to accounts token position
+    max_settle_mngo = max_settle_mngo.min(account_mngo_native);
+    max_settle = cm!(max_settle_mngo * mngo_oracle_price / discount_rate);
```

Comment:
If we stored the inverse of discount rate, we could skip one division:
```
let mngo_settle_price = cm!(mngo_oracle_price * price_bonus);`
```
and then divide and multiply with mngo_settle_price instead